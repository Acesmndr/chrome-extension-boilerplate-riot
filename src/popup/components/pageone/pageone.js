import { sendMessage } from '../../../chrome/chrome-utils';

export default {
  goToSecondPage() {
    this.route('pagetwo', {});
  },
  getAjaxDataFromBackground() {
    sendMessage({ type: 'sendAjaxRequest' });
  },
  onMounted() {
    console.log("First page has been mounted");
  },
};
