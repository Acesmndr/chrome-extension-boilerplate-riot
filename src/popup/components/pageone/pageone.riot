<pageone>
	<div class='card'>
		<div class='content'>
			<div class='graphic'>
				<div class='title'>Riot Chrome Extension Boilerplate</div>
				<img class='illustration' src='assets/img/illustration.svg' />
				<div class='author'>acesmndr@gmail.com</div>
			</div>
		</div>
		<div class='content right-align'>
			<input type='button' class='request-btn' onclick={ getAjaxDataFromBackground } value='Send Data Request'/>
			<input type='button' onclick={ goToSecondPage } value='Go to page 2'/>
		</div>
	</div>
	<style type='scss'>
		@import './src/popup/components/pageone/pageone';
	</style>
	<script>
		import script from './pageone.js';
		export default script;
	</script>
</pageone>
