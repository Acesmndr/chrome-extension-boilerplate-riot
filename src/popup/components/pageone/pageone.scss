@import '../../style/colorpalette';

.graphic {
  padding: 4px;
  color: grey;
  text-align: center;
  .title {
    font-size: 20px;
  }
  .illustration {
    width: 300px;
    margin: 24px 0 8px;
  }
  .author {
    text-align: right;
    font-size: 8px;
    opacity: 0.5;
  }
}

.request-btn {
  background-color: $pinkish-red;
}
