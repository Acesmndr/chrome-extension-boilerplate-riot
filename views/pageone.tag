<pageone>
	<div class='card'>
		<div class='content'>
			<div class='graphic'>
				<div class='huge'>RIOT CHROME EXTENSION BOILERPLATE</div>
				<div class='medium'>-acesmndr</div>
			</div>
		</div>
		<div class='content right-align'>
			<input type='button' onclick={ goToSecondPage } value='Go to page 2'/>
		</div>
	</div>
	<style type='scss'>
		@import 'styles/pageone';
	</style>
	<script src="presenters/pageone.js"></script>
</pageone>