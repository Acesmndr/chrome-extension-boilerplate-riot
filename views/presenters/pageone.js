this.pageRedirect = () => {
  riot.mount('.main-body', 'pagetwo');
}