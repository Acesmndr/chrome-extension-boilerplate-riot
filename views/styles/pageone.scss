@import './colorpalette';
.graphic {
  background: #d32978;
  padding: 16px;
  color: white;
  // margin: -16px -16px 0px -16px;
  .huge {
    font-size: 62px;
  }
  .medium {
    text-align: right;
    font-size: 32px;
    opacity: 0.2;
  }
}