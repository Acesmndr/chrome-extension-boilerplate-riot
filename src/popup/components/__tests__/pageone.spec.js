import * as riot from 'riot';
import pageone from '../pageone/pageone.riot';

riot.register('pageone', pageone);

describe('Page one specs', () => {
    beforeAll( () => {
        const elem = document.createElement('div');
        elem.className = 'main-body';
        document.body.appendChild(elem)
        riot.mount(elem, {}, 'pageone');
    });
    
    
    it('should mount the tag', () => {
        expect(document.querySelector('.main-body')).toMatchSnapshot();
    });

    it('should have a title text', () => {
        expect(document.querySelector('.graphic .title').textContent).toBe('Riot Chrome Extension Boilerplate');
        expect(document.querySelector('.graphic .author').textContent).toBe('acesmndr@gmail.com');
    });

    it('should have two buttons', () => {
        expect(document.querySelectorAll('input[type="button"]').length).toBe(2);
    });
});