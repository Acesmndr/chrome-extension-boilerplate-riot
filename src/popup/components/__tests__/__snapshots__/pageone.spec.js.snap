// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page one specs should mount the tag 1`] = `
<div
  class="main-body"
  is="pageone"
>
  <div
    class="card"
  >
    <div
      class="content"
    >
      <div
        class="graphic"
      >
        <div
          class="title"
        >
          Riot Chrome Extension Boilerplate
        </div>
        <img
          class="illustration"
          src="assets/img/illustration.svg"
        />
        <div
          class="author"
        >
          acesmndr@gmail.com
        </div>
      </div>
    </div>
    <div
      class="content right-align"
    >
      <input
        type="button"
        value="Send Data Request"
      />
      <input
        type="button"
        value="Go to page 2"
      />
    </div>
  </div>
</div>
`;
